platform_plugin_aspects/views.py

```diff
@@ -122,3 +131,64 @@ def get(self, request, *args, **kwargs):
             raise APIException() from exc
 
         return Response({"guestToken": guest_token})
+
+
+class SupersetEmbeddedDashboardView(GenericAPIView):
+    """
+    Superset endpoint for embedded dashboard (for in-context analytics) data.
+    """
+
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    lookup_field = "usage_key_string"
+
+    def get_object(self):
+        """
+        Return a usage key or course key for the requested usage_key.
+        """
+        key_string = self.kwargs.get(self.lookup_field, "")
+        try:
+            usage_key = UsageKey.from_string(key_string)
+        except InvalidKeyError as exc:
+            try:
+                usage_key = CourseKey.from_string(key_string)
+            except InvalidKeyError:
+                raise NotFound(
+                    _("Invalid usage key: '{key_string}'").format(key_string=key_string)
+                ) from exc
+        return usage_key
+
+    def get(self, request, *args, **kwargs):
+        """
+        Return Superset context for embedding the dashboard for the requested block.
+        """
+        usage_key = self.get_object()
+        if isinstance(usage_key, CourseKey):
+            block_type = "course"
+        else:
+            block_type = usage_key.block_type
+
+        dashboards = settings.ASPECTS_EMBEDDED_DASHBOARDS
+        dashboard = dashboards.get(block_type)
+        if dashboard is None:
+            raise NotFound(
+                _("No dashboard for block type: '{block_type}'").format(block_type=block_type)
+            )
+
+        if dashboard.get("allow_translations"):
+            user = get_current_user()
```

Comment:
I think you don't need `get_current_user` from `crum` here and can use just `request.user`.